go/worker/sandbox/containerdPool.go

```diff
@@ -0,0 +1,311 @@
+        "path/filepath"
+        "sync/atomic"
+        "syscall"
+        "strings"
+        "time"
+
+        "github.com/containerd/containerd"
+        "github.com/containerd/containerd/cio"
+        "github.com/containerd/containerd/namespaces"
+        "github.com/containerd/containerd/oci"
+        "github.com/open-lambda/open-lambda/go/common"
+        "github.com/open-lambda/open-lambda/go/worker/sandbox/containerdutil"
+        "github.com/containerd/containerd/snapshots"
+        "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+// ContainerdPool is a SandboxPool that creates containerd containers.
+type ContainerdPool struct {
+        client        *containerd.Client
+        namespace     string
+        runtime       string
+        image         string
+        labels        map[string]string
+        idxPtr        *int64
+        eventHandlers []SandboxEventFunc
+        debugger
+}
+
+// NewContainerdPool creates a ContainerdPool.
+func NewContainerdPool() (*ContainerdPool, error) {
+        client, err := containerd.New(common.Conf.Containerd.SocketAddress,
+                containerd.WithDefaultNamespace(common.Conf.Containerd.Namespace),
+                containerd.WithTimeout(10*time.Second),
+        )
+        if err != nil {
+                return nil, fmt.Errorf("failed to connect to containerd: %v", err)
+        }
+
+        ctx := namespaces.WithNamespace(context.Background(), common.Conf.Containerd.Namespace)
+
+        // ensure that containerd is serving
+        healthCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
+        defer cancel()
+        serving, err := client.IsServing(healthCtx)
+        if err != nil {
+                return nil, fmt.Errorf("failed to check containerd health: %v", err)
+        }
+        if !serving {
+                return nil, fmt.Errorf("containerd is not serving at %s", common.Conf.Containerd.SocketAddress)
+        }
+        slog.Info("Successfully connected to containerd", "socket_address", common.Conf.Containerd.SocketAddress, "namespace", common.Conf.Containerd.Namespace)
+        var sharedIdx int64 = -1
+        pool := &ContainerdPool{
+                client:    client,
+                namespace: common.Conf.Containerd.Namespace,
+                runtime:   common.Conf.Containerd.Runtime,
+                image:     common.Conf.Containerd.Base_image,
+                labels: map[string]string{
+                        containerdutil.CONTAINERD_LABEL_CLUSTER: common.Conf.Worker_dir,
+                        containerdutil.CONTAINERD_LABEL_RUNTIME: common.Conf.Containerd.Runtime,
+                },
+                idxPtr:        &sharedIdx,
+                eventHandlers: []SandboxEventFunc{},
+        }
+
+        pool.debugger = newDebugger(pool)
+
+        return pool, nil
+
+}
+
+
+func (pool *ContainerdPool) Create(parent Sandbox, isLeaf bool, codeDir, scratchDir string, meta *SandboxMeta, rtType common.RuntimeType) (sb Sandbox, err error) {
```

Comment:
I think this function needs some more explanation in the comments.  Don't go too heavy on comments, but need to explain a bit about namespaces, tasks, NoNewPrivileges, etc.  Feel free to link to containterd documentation as appropriate to backup brief claims.